indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -309,6 +309,26 @@ public static void injectSystemProperties(Job job)
   {
     injectSystemProperties(job.getConfiguration());
   }
+  public static void injectDruidProperties(Configuration configuration, List<String> listOfAllowedPrefix) {
+    String mapJavaOpts = configuration.get(MRJobConfig.MAP_JAVA_OPTS);
+    String reduceJavaOpts = configuration.get(MRJobConfig.REDUCE_JAVA_OPTS);
+
+    for (String propName: System.getProperties().stringPropertyNames()){
+      for (String prefix : listOfAllowedPrefix) {
+        if (propName.startsWith(prefix)) {
+          mapJavaOpts = String.format("%s -D%s=%s", mapJavaOpts, propName, System.getProperty(propName));
+          reduceJavaOpts = String.format("%s -D%s=%s", reduceJavaOpts, propName, System.getProperty(propName));
+        }
```

Comment:
Would be better to put a `break` in here if you get a match so that it doesn't add the same property multiple times if multiple prefixes match.